python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py

```diff
@@ -305,6 +228,30 @@ def _overlay_her_flood_lists(self: AvdStructuredConfigNetworkServicesProtocol) -
 
         return overlay_her_flood_lists
 
-    @cached_property
-    def _multi_vtep(self: AvdStructuredConfigNetworkServicesProtocol) -> bool:
-        return self.shared_utils.mlag is True and self.shared_utils.evpn_multicast is True
+    def _check_for_duplicates(
+        self: AvdStructuredConfigNetworkServicesProtocol,
+    ) -> None:
+        """
+        Raise an exception if the passed vlan or VRF object VNI is already in the vnis dictionary.
+
+        If no duplicate is found, add the vlan or vrf object to the vnis dictionary
+        """
+        vnis: dict[int, EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.VrfsItem | EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.VlansItem] = {}
+        for vlan in self.structured_config.vxlan_interface.vxlan1.vxlan.vlans:
+            if vlan.vni in vnis:
+                conflicting_object_type = "VLAN" if isinstance(vnis[vlan.vni], EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.VlansItem) else "VRF"
+                msg = (
+                    "Found duplicate objects with conflicting data while generating configuration for VXLAN VNIs. "
+                    f"VLAN {vlan._as_dict()} conflicts with {conflicting_object_type} {vnis[vlan.vni]._as_dict()}."
```

Comment:
Maybe we could make these messages more clear, instead of dumping the full objects.
```
Found conflicting VNI 321 between VLAN 123 ('MY_VLAN_NAME') in Tenant 'A' and VRF 'foo' in Tenant 'B'.
```
The first tenant part should only be printed if the second is different. I think we have (or had) this logic somewhere.

Reply:
at this time I don't hve any info about the tenant when I go through this

Reply:
fixed but this is bringing some new changes in negative unit test where we were not consifering tenant before (which was, granted, less helpful)